apps/datapuller/src/section.ts

```diff
@@ -0,0 +1,67 @@
+import { ISectionItem, SectionModel } from "@repo/common";
+import { ClassSection, ClassesAPI } from "@repo/sis-api/classes";
+
+import setup from "./shared";
+import { Config } from "./shared/config";
+import mapSectionToNewSection from "./shared/sectionParser";
+import { fetchActiveTerms, fetchPaginatedData } from "./shared/utils";
+
+async function updateSections(config: Config) {
+  const log = config.log;
+  const classesAPI = new ClassesAPI();
+
+  log.info("Fetching Active Terms");
+  const activeTerms = await fetchActiveTerms(log, {
+    app_id: config.sis.TERM_APP_ID,
+    app_key: config.sis.TERM_APP_KEY,
+  });
+
+  log.info(activeTerms);
+
+  const sections = await fetchPaginatedData<ISectionItem, ClassSection>(
+    log,
+    classesAPI.v1,
+    activeTerms,
+    "getClassSectionsUsingGet",
+    {
+      app_id: config.sis.CLASS_APP_ID,
+      app_key: config.sis.CLASS_APP_KEY,
+    },
+    (data) => data.apiResponse.response.classSections || [],
+    mapSectionToNewSection
+  );
+
+  log.info("Example Section:", sections[0]);
+
+  log.info(`Updated ${sections.length} sections for Spring 2024`);
+
+  await SectionModel.deleteMany({});
```

Comment:
You should only delete sections you are going to add. This should look like:
```ts
await SectionModel.deleteMany({
  "class.session.term.id": { "$in": activeTerms }
});
```
(if `activeTerms` is an array of term ids)